pkg/buildkite/builds.go

```diff
@@ -565,3 +714,39 @@ func convertEntries(entries []Entry) map[string]string {
 	}
 	return result
 }
+
+func getTimestampStringOrNil(ts *buildkite.Timestamp) *string {
+	if ts == nil {
+		return nil
+	}
+	str := ts.Format(time.RFC3339)
+	return &str
+}
+
+// see https://buildkite.com/docs/pipelines/configure/notifications#build-states
+func isTerminalState(state string) bool {
+	switch state {
+	case "passed", "failed", "blocked", "canceled":
```

Comment:
I'd double check this list in bk/bk to make sure you cover timed out / expired

Reply:
Actually those are job statuses I think not build statuses now I think of it. You can see all the build states here though: https://github.com/buildkite/buildkite/blob/main/app/models/build.rb#L75